aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -125,6 +125,36 @@ Tensor sparse_coo_tensor(ArrayRef<int64_t> size, const TensorOptions& options) {
   return at::_sparse_coo_tensor_with_dims(size.size(), 0, size, options.layout(at::kSparse));
 }
 
+/*eye init*/
+Tensor& eye_sparse(Tensor& result, int64_t n) {
+  return eye_sparse(result, n, -1);
+}
+
+Tensor& eye_sparse(Tensor& result, int64_t n, int64_t m) {
+  TORCH_CHECK(n >= 0, "n must be greater or equal to 0, got ", n);
+
+  if(m < 0) {
+    m = n;
+  }
+
+  // take the smaller of the rows/columns
+  int64_t sz = (m < n) ? m : n;
+
+  Tensor values = at::ones({sz});
+  // hard code sparse dimensions to 2 as this is a guranteed matrix
+  Tensor indices = at::empty({2, sz});
```

Comment:
You can do something like `indices = torch.arange(sz, options).unsqueeze(0).expand(2,-1)`, accessing by element as you do below is very inefficient.

Reply:
Have incorporated the suggestions